src/brogue/Combat.c

```diff
@@ -325,6 +325,13 @@ void moralAttack(creature *attacker, creature *defender) {
         }
         defender->status[STATUS_ENTRANCED] = 0;
 
+        boolean adjacent;
+        adjacent = distanceBetween(attacker->xLoc, attacker->yLoc, defender->xLoc, defender->yLoc) <= 1 && ((attacker->xLoc == defender->xLoc) || (attacker->yLoc == defender->yLoc));
```

Comment:
```
##g.
.@..
##gg
```

In a situation like this, if the player throws a weapon or fires a staff at either of the goblins diagonal to him, that goblin should emerge from the room and attack him.

```
##g.
.,@.
##gg
```

You are correct that if the player stepped here and attacked the lower rightmost goblin with a melee weapon, that goblin would follow him out of the room (when he probably shouldn't).

To make the goblins even smarter, we should check for the impassibility of the surrounding hexes, so we know if the player is attacking the goblin from a space where a melee attack cannot be made.

Reply:
I'm not too keen on this, it seems to make the behaviour more inconsistent and hard to predict. Doing things based on diagonals because goblins stand diagonally when guarding corridors... Perhaps we should think about how this behaviour should work in general, e.g. attacking a goblin in a room and then fleeing, should they chase? Etc.